### PR TITLE
[GPU] Fix cl kernel build error

### PR DESCRIPTION
**Pull request by ahnyoung-paul**
### Details:
 - *Fix cl kernel build error*
 - *Create activation jit constant using output data type instead unit type*

### Tickets:
 - *96493*

### CODE REVIEW

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/strided_slice_ref.cl [hunk 1]
```diff
@@ -326,7 +326,7 @@ KERNEL(strided_slice_ref)(OPTIONAL_SHAPE_INFO_ARG
     FUSED_OPS;
     output[output_index] = FUSED_OPS_RESULT;
 #else
-    output[output_index] = ACTIVATION(input[input_index], ACTIVATION_PARAMS);
+    output[output_index] = ACTIVATION((UNIT_TYPE)input[input_index], (UNIT_TYPE)ACTIVATION_PARAMS);
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/strided_slice_ref.cl:329]: Don't use unit type. Cast to either OUTPUT_TYPE or some kind of ACCUMULATOR_TYPE

Comment (ahnyoung-paul) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/strided_slice_ref.cl]: I'll fix it. thanks.

Comment (ahnyoung-paul) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/strided_slice_ref.cl]: Fixed it. please review it again.

**Comment by andrew-k-park** [comment]
Could you add a test to cover this case?

File: src/plugins/intel_gpu/src/kernel_selector/kernel_base.cpp [hunk 2]
```diff
@@ -87,7 +87,7 @@ JitConstants KernelBase::MakeBaseParamsJitConstants(const base_params& params, b
 
     // for activation function
     jit.Merge(MakeUnitTypeJitConstants(unitType));
-    jit.Merge(MakeActivationJitConstants(params.activations, unitType));
+    jit.Merge(MakeActivationJitConstants(params.activations, params.outputs[0].GetDType()));
```
Comment (ahnyoung-paul) [inline src/plugins/intel_gpu/src/kernel_selector/kernel_base.cpp:90]: Need to add comment why data type is changed from unit type to output data type

**Comment by ahnyoung-paul** [comment]
> Could you add a test to cover this case?

Done

**Review by andrew-k-park — APPROVED** [review]

**Comment by andrew-k-park** [comment]
Could you also check failed fusings_gpu TCs for ov_gpu_unit_tests?

**Review by e-ddykim — APPROVED** [review]

**Pull request merged**